From 897b638e98fc2c210d3b5f1dbc6cbca72e444b22 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:48 -0400
Subject: [scsi] qla2xxx: IOCB data should be copied to I/O mem using memcpy_toio

Message-id: <1408391817-6389-61-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87959
O-Subject: [PATCH RHEL 7.1 e-stor 60/69] qla2xxx: IOCB data should be copied to I/O mem using memcpy_toio.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Atul Deshmukh <atul.deshmukh@qlogic.com>

scsi commit id f79a84bfafa53c5ff5acb1b1099b7223ab7c5fba

Signed-off-by: Atul Deshmukh <atul.deshmukh@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 4bbeb63..948b8d8 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -3425,7 +3425,7 @@ qlafx00_fxdisc_iocb(srb_t *sp, struct fxdisc_entry_fx00 *pfxiocb)
      sp->fcport->vha, 0x3047,
      (uint8_t *)&fx_iocb, sizeof(struct fxdisc_entry_fx00));
 
- memcpy((void *)pfxiocb, &fx_iocb,
+ memcpy_toio((void __iomem *)pfxiocb, &fx_iocb,
      sizeof(struct fxdisc_entry_fx00));
  wmb();
 }
-- 
1.7.1